### ISSUE TITLE
`min_tokens_to_keep` has to be a positive integer, but it's type is float in mlx_lm/generate.py

### ISSUE THREAD

**Issue by asamiKA** [ml-explore/mlx-examples#1134]
In mlx_lm/generate.py, the type of `--min-tokens-to-keep` parameter is set to `float`.

https://github.com/ml-explore/mlx-examples/blob/0ca162cfb2a85164cbec18304913c4220520786c/llms/mlx_lm/generate.py#L80
```python
    parser.add_argument(
        "--min-tokens-to-keep",
        type=float,  # <- int
        default=DEFAULT_MIN_TOKENS_TO_KEEP,
        help="Minimum tokens to keep for min-p sampling.",
    )
```

but in mlx_lm/sample_utils.py, this cause error that '`min_tokens_to_keep` has to be a positive integer'.
https://github.com/ml-explore/mlx-examples/blob/0ca162cfb2a85164cbec18304913c4220520786c/llms/mlx_lm/sample_utils.py#L109

```python
    if not (0 <= min_p <= 1.0):
        raise ValueError(
            f"`min_p` has to be a float in the [0, 1] interval, but is {min_p}"
        )
    if not isinstance(min_tokens_to_keep, int) or (min_tokens_to_keep < 1):
        raise ValueError(
            f"`min_tokens_to_keep` has to be a positive integer, but is {min_tokens_to_keep}"
        )
   ```

**Comment by awni** [comment]
Good catch, that's a typo. Would you like to send a PR to fix it?

**Comment by awni** [comment]
Actually I just added the fix in #1133
